src/os_sensor_nodelet.cpp

```diff
@@ -313,6 +317,13 @@ sensor::sensor_config OusterSensor::parse_config_from_ros_parameters() {
         config.udp_port_imu = imu_port;
     }
 
+    persist_config = nh.param("persist_config", false);
+    if (persist_config && (lidar_port == 0 || imu_port == 0)) {
+        NODELET_WARN("When using persist_config it is recommended to not "
+        "use 0 for port value");
+    }
```

Comment:
I am not sure if this warning would be necessary, open to your suggestions

Reply:
If only the FW would let you change ports without a reinit, then all our problems would go away.....

I think the warning is fine, maybe could make the language more specific.

"When using persist_config it is recommended to not use 0 for port value as this requires reinitializing the sensor each time anyways."

Reply:
> If only the FW would let you change ports without a reinit, then all our problems would go away.

Unfortunately it doesn't allow that :face_with_diagonal_mouth:

Reply:
Maybe we can use this as an example for making the case to allow that

Reply:
> As a crazy idea, it is probably possible to look at the current sensor port, see if it is available and just use that rather than grabbing a new ephemeral port. That way we can mostly avoid reinits.

I like this suggestion.

Reply:
> As a crazy idea, it is probably possible to look at the current sensor port, see if it is available and just use that rather than grabbing a new ephemeral port. That way we can mostly avoid reinits.

It isn't crazy to me, however, this needs further work. I'd suggest to achieve this in a follow up ticket and have the core sdk provide this functionality (check if the current ports accessible) then we can use here.